From a8cf86e53a7e56a4e52a936fa2c23d1209e839bb Mon Sep 17 00:00:00 2001
From: Amos Kong <akong@redhat.com>
Date: Wed, 6 Aug 2014 05:16:21 -0400
Subject: [virt] virtio-rng: fix boot with virtio-rng device

Message-id: <1407302182-1492-4-git-send-email-akong@redhat.com>
Patchwork-id: 87483
O-Subject: [RHEL-7.1 kernel PATCH 3/4] virtio-rng: fix boot with virtio-rng device
Bugzilla: 915335
RH-Acked-by: Jason Wang <jasowang@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

From: Sasha Levin <sasha.levin@oracle.com>

Commit "virtio-rng: support multiple virtio-rng devices" has broken
boot with a virtio-rng device because the 'init' callback of the
virtio-rng device was left unitialized to garbage, and got called
by the hwrng infrastructure, killing the guest on boot.

Signed-off-by: Sasha Levin <sasha.levin@oracle.com>
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
Fixes: 08e53fbdb85c0f6f45c0f7c1ea3defc1752a95ce
(cherry picked from commit e5d23a8cc38c555f9f3a40dcc5d14030105df9a8)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/char/hw_random/virtio-rng.c b/drivers/char/hw_random/virtio-rng.c
index ac16ab9..7b7d93a 100644
--- a/drivers/char/hw_random/virtio-rng.c
+++ b/drivers/char/hw_random/virtio-rng.c
@@ -95,7 +95,7 @@ static int probe_common(struct virtio_device *vdev)
  int err, i;
  struct virtrng_info *vi = NULL;
 
- vi = kmalloc(sizeof(struct virtrng_info), GFP_KERNEL);
+ vi = kzalloc(sizeof(struct virtrng_info), GFP_KERNEL);
  vi->hwrng.name = kmalloc(40, GFP_KERNEL);
  init_completion(&vi->have_data);
 
-- 
1.7.1